apprentice_teacher.rb

```diff
@@ -1,21 +1,10 @@
-class ApprenticeTeacher
-  attr_reader :age, :salary, :phase, :target_raise
-  attr_accessor :name
+require_relative 'educator'
+require_relative 'high_five'
+class ApprenticeTeacher < Educator
 
   def initialize(options={})
-    @age = options.fetch(:age, 0)
-    @name = options.fetch(:name, "")
+    super
     @target_raise = 800
-    @phase = 3
-  end
-
-  def offer_high_five
-    "High five!"
-  end
-
-  def set_phase(num)
-    @phase = num
-    "Cool, I've always wanted to teach phase #{num}!"
   end
 
   def teach_stuff
```

Comment:
this method shows up twice. there are a few ways to approach abstracting this. Here's one cool way: 
```
  def teach_stuff
    staff = self.class.name
    response = ""
    response += "Listen, class, this is how everything works"
    case staff
    when "Teacher"
      response += ", fo SHO! "
      response += "*drops flat-out insane knowledge bomb* "
      response += "... You're welcome. *saunters away*"
    when "ApprenticeTeacher"
      response += ". *drops crazy knowledge bomb* "
      response += "... You're welcome."
    end
    response
  end
```